metrics-tracing-context/src/tracing_integration.rs

```diff
@@ -35,46 +44,45 @@ impl Default for Labels {
 impl Visit for Labels {
     fn record_str(&mut self, field: &Field, value: &str) {
         let label = Label::new(field.name(), value.to_string());
-        self.0.push(label);
+        self.0.insert(field.name().into(), label);
```

Comment:
We should either:

- use `&'static str' as the map key type since we don't use the key for anything other than deduplicating
- keep the key as `SharedString` but then change the map value type to `SharedString` and construct the `Label` from the key/value in `enhance_key`

Reply:
Went with the first option